### PR TITLE
treewide: fix redirected urls

### PR DESCRIPTION
**Pull request by Synthetica9**
###### Motivation for this change

Related:
 - 9fc5e7e
 - 593e11f
 - 508ae42

Since the last time I ran this script, the Repology API changed, so I had to
adapt the script used in the previous PR. The new API should be more robust, so
overall this is a positive (no more grepping the error messages for our relevant
data but just a nice json structure).

Here's the new script I used:

```sh
curl repology.org/api/v1/repository/nix_unstable/problems \
   | jq -r '.[] | select(.type == "homepage_permanent_https_redirect") | .data | "s@\(.url)@\(.target)@"' \
   | sort | uniq | tee script.sed
find -name '*.nix' | xargs -P4 -- sed -f script.sed -i
```

I also added this script to `maintainers/scripts`.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Synthetica9** [comment]
Sorry for the pings code owners :(

File: pkgs/tools/security/pcsc-scm-scl011/default.nix [hunk 1]
```diff
@@ -39,7 +39,7 @@ stdenv.mkDerivation rec {
 
   meta = with stdenv.lib; {
     description = "SCM Microsystems SCL011 chipcard reader user space driver";
-    homepage = "http://www.scm-pc-card.de/index.php?lang=en&page=product&function=show_product&product_id=630";
+    homepage = "https://www.scm-pc-card.de/index.php?lang=enhttp://www.scm-pc-card.de/index.php?lang=en&page=product&function=show_product&product_id=630page=producthttp://www.scm-pc-card.de/index.php?lang=en&page=product&function=show_product&product_id=630function=show_producthttp://www.scm-pc-card.de/index.php?lang=en&page=product&function=show_product&product_id=630product_id=630";
```
Comment (marsam) [inline pkgs/tools/security/pcsc-scm-scl011/default.nix:42]: I think it failed here

Comment (Synthetica9) [reply pkgs/tools/security/pcsc-scm-scl011/default.nix]: Huh, probably an oddity of the particular script used. Fixed it manually

**Review by marsam — APPROVED** [review]

**Review by jonringer — APPROVED** [review]
diff LGTM
nice script :)

was able to run ofborg locally
```
$ nix-instantiate ./nixos/release.nix --option restrict-eval true -I foo=. --arg nixpkgs '{ outPath = ./.; revCount = 0; shortRev = "aabbcc"; }'
warning: dumping very large path (> 256 MiB); this may run out of memory
warning: you did not specify '--add-root'; the result might be removed by the garbage collector
/nix/store/bzhayi9ljcy432z34lj9hlz1n151i2ax-nixos-channel-21.03pre0.aabbcc.drv
/nix/store/mdca90jpg2d8w6k3if565rp3vrd3wllv-options.json.drv
```

**Pull request merged**